app/Http/Controllers/Auth/RegisterController.php

```diff
@@ -41,33 +44,29 @@ public function __construct()
         $this->middleware('guest');
     }
 
-    /**
-     * Get a validator for an incoming registration request.
-     *
-     * @param  array  $data
-     * @return \Illuminate\Contracts\Validation\Validator
-     */
-    protected function validator(array $data)
-    {
-        return Validator::make($data, [
-            'name' => ['required', 'string', 'max:255'],
-            'email' => ['required', 'string', 'email', 'max:255', 'unique:users'],
-            'password' => ['required', 'string', 'min:8', 'confirmed'],
-        ]);
-    }
-
     /**
      * Create a new user instance after a valid registration.
      *
      * @param  array  $data
      * @return \App\Models\User
      */
-    protected function create(array $data)
+    
+    protected function create(Request $data)
     {
         return User::create([
             'name' => $data['name'],
             'email' => $data['email'],
             'password' => Hash::make($data['password']),
+            'role' => User::ROLE_STUDENT,
         ]);
     }
+    
+    public function register(RegisterRequest $request)
+    {
+        $registerRequest = $this->create($request);
```

Comment:
`$this->create($request);` này trả ra gì nhỉ?

Reply:
create này trả về một array chứa các thông tin đã nhập ở phần register ạ

Reply:
đối tượng ở đây được đề cập đến là gì? tạo ra đối tượng gì?